lib/elements/elements.dart

```diff
@@ -505,3 +505,49 @@ class _ElementTextProperty implements Property {
   // TODO:
   Stream get onChanged => null;
 }
+
+class TabController {
+  List<TabElement> tabElements;
+
+  TabElement _selectedTab;
+
+  TabController({this.tabElements}) {
+    if (tabElements != null) {
+      tabElements.forEach((tab) => registerTab(tab));
+    } else {
+      tabElements = [];
+    }
+  }
+
+  TabElement get selectedTab => tabElements.firstWhere((tab) => tab.hasAttr("selected"));
+
+  set selectedTab(TabElement s) {
+    _selectedTab = s;
+    tabElements.where((tab) => tab != _selectedTab).forEach((tab) {
+      tab.element.attributes.remove('selected');
+    });
+    _selectedTab.setAttr('selected');
+  }
+
+  void registerTab(TabElement tab) {
+    tabElements.add(tab);
+    tab.onClick.listen((_) {
+      selectedTab = tab;
+      tab.onSelect();
+    });
+  }
+
+  void selectTab(String tabName) {
+    selectedTab = tabElements.firstWhere((t) => t.name == tabName);
+  }
+}
+
+class TabElement extends DElement {
+  String name;
+
+  Function onSelect;
```

Comment:
I do the onSelect after initialization:

```
      controller.registerTab(new TabElement(querySelector('#${name}tab'), name: name)
        ..onSelect = () {
          Element issuesElement = querySelector('#issues');
          if (name != "dart") {
            issuesElement.style.display = "none";
          } else {
            issuesElement.style.display = "block";
          }
          ga.sendEvent('edit', name);
          _context.switchTo(name);
        }
      );
```

To make it more readable, I could change that, and make this field final then ?